File: app/widgets/molecular/primers/marker_auto_complete_field.js
////= require <templates/auto_complete_field>
////= require <molecular/marker>
//
//JooseModule('Molecular.Primers', function () {
//  JooseClass('MarkerAutoCompleteField', {
//    isa: Templates.AutoCompleteField,
//    has: {
//      primer: { is: 'ro', required: true, nullable: false },
//      object: { is: 'ro', init: function () { return this.primer() } },
//      method:        { is: 'ro', init: 'marker' },
//      valueMethod:   { is: 'ro', init: 'id' },
//      textMethod:    { is: 'ro', init: 'name' },
//      width:         { is: 'ro', init: 275 },
//      collection: { is: 'ro', init: function () { return (
//        Molecular.Marker.
//          collection({ context: this.context() }).
//          select('id', 'name')
//      ) } }
//    }
//  })
//})

File: app/widgets/molecular/primers/marker_auto_complete_field.rb
class Molecular::Primers::MarkerAutoCompleteField < Templates::AutoCompleteField
  def initialize options
    @primer ||= options[:primer] || fail('primer not given')
    @model_object ||= @primer
    @attribute_path ||= 'marker'
    @value_method = 'id'
    @text_method = 'name'
    @width = 275
    super
  end
end

